File: Week1/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    int coinsUsed = 0;
    float change;
    do
    {
        change = get_float("Change owed?\n"); //This is a propmt that asks the user how much change is owed.
    }
    while (change < 0); //This is a loop that keeps, asking the question if the user gives a negative number.

    int cents = round(change * 100); //This turns the value the user gave in dollars to cents.
    
    while (cents >= 25) //This checks how many quarters fit in the change.
    {
        cents -= 25;
        coinsUsed++;
    }
    while (cents >= 10) //This checks how many dimes still fit in the change.
    {
        cents -= 10;
        coinsUsed++;
    }
    while (cents >= 5) //This checks how many nickels still fit in the change.
    {
        cents -= 5;
        coinsUsed++;
    }
    while (cents >= 1) //This checks how many pennies still fit in the change. 
    {
        cents -= 1;
        coinsUsed++;
    }
    printf("%i\n", coinsUsed);
}

File: week6/cash.py
from cs50 import get_float

counter = 0

change = -1
while change < 0:
    change = get_float("Change owed: ")
    
cents = round(change * 100)

while cents >= 25:
    cents -= 25
    counter += 1

while cents >= 10:
    cents -= 10
    counter += 1

while cents >= 5:
    cents -= 5
    counter += 1

while cents >= 1:
    cents -= 1
    counter += 1

print(counter)
    
